### PR TITLE
Use `disable_decoding` in async `read_response` with hiredis parser.

### LINKED ISSUE #3035: `disable_decoding` not used in async version of `read_response`.
**Version**: redis-py version - 5.0.0 (I still see it on main)

**Description**:
Is there a reason why `disable_decoding` is not being used in async version of `read_response`? 
It seems like there's no async way to get raw data (`NEVER_DECODE`) from Redis.
Are there any plans of adding the `NEVER_DECODE` support for async `read_response`?
We want to save pickled data (not compatible with `UTF-8`) in Redis, and for now we will use the workaround (as in https://github.com/redis/redis-py/issues/1006#issuecomment-1100506453 ), but it's not a very clean solution.

redis/_parsers/hiredis.py:204
https://github.com/redis/redis-py/blob/d3a3ada03e080f39144807c9fbe44876c40e0548/redis/_parsers/hiredis.py#L192-L217

### PR DESCRIPTION
**Pull request by MrDenkoV**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Do tests and lints pass with this change?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [ ] Is the new or changed code fully tested?
- [x] ~~Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?~~
- [x] ~~Is there an example added to the examples folder (if applicable)?~~
- [x] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

Fix #3035 
Use `read_response` in async `read_response`.

### CODE REVIEW

**Review by dvora-h — APPROVED** [review]

File: redis/_parsers/hiredis.py [hunk 1]
```diff
@@ -198,10 +198,16 @@ async def read_response(
         if not self._connected:
             raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR) from None
 
-        response = self._reader.gets()
+        if disable_decoding:
+            response = self._reader.gets(False)
+        else:
+            response = self._reader.gets()
```
Comment (kristjanvalur) [inline redis/_parsers/hiredis.py:204]: why not `response=self._reader.gets(not disable_decoding)`
or
```python
enable_decoding = not disable decoding
response=self._reader.gets(enable_decoding)
```

Comment (MrDenkoV) [reply redis/_parsers/hiredis.py:204]: It's been a while so I don't exactly remember, especially that there wasn't any sophisticated reason.
I think I wanted to make it consistent with the sync version + I had a problem to get to the `gets` definition, and didn't want to take much more time digging deeper.

Comment (kristjanvalur) [reply redis/_parsers/hiredis.py:204]: Yes, well, I had a look at the source:
```c
 self->shouldDecode = 1;
    if (!PyArg_ParseTuple(args, "|i", &self->shouldDecode)) {
        return NULL;
    }
```
Basically, a `shouldDecode` argument can just be a Python bool.
so a `should_decode = not disable_decoding` would have been simpler and more readable.  

Anyway, just came across this, don't worry.

Comment (MrDenkoV) [reply redis/_parsers/hiredis.py:204]: Thank you for pointing that out, and linking the code! 😄

**Pull request merged**